modules/schema/src/main/java/org/apache/ignite/internal/util/ObjectFactory.java

```diff
@@ -35,17 +36,25 @@ public class ObjectFactory<T> implements Factory<T> {
      * @throws IllegalArgumentException If no default constructor found.
      */
     public ObjectFactory(Class<T> clazz) {
-        try {
-            cnstr = clazz.getDeclaredConstructor();
-            cnstr.setAccessible(true);
-        } catch (NoSuchMethodException e) {
-            throw new IllegalArgumentException("Class has no default constructor: class=" + clazz.getName(), e);
+        if (clazz == Void.class) {
+            cnstr = null;
```

Comment:
But is it really necessary? I mean, is it documented anywhere?
Maybe null value always means "no" value and we introduce a bug here?

Reply:
I couldn't find any documentation and the tests pass, so... I'll ask the author of this code just in case